enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,782 @@
+we expect the conversion to happen much in the same way as a regular MachineSet.
+
+ControlPlaneMachineSets (additionally to MachineSets) have a concept of failure domains.
+This is also a concept in Cluster API.
+When converting ControlPlaneMachineSets to Cluster API, we will ensure that the appropriate failure domain configuration is configured inside the Cluster API InfrastructureCluster,
+and that the failure domains from the ControlPlaneMachineSet Machine API configuration are reflected across.
+
+When deploying Machines using the future Cluster API extension, the InfrastructureTemplate will be combined with the named failure domains on the InfrastructureCluster
+to spread the Machines across multiple failure domains.
+
+#### Creation of new Machines
+
+When a Machine API MachineSet creates a new Machine, the sync controller will create the equivalent Cluster API Machine based on the method described in [Machine Synchronisation](#machine-synchronisation).
+
+When a Cluster API MachineSet creates a new Machine, the sync controller will create its Machine API equivalent if, and only if, the Cluster API MachineSet has a Machine API equivalent.
+This will allow users to switch a MachineSet back to Machine API at a later date if they wish to do so.
+
+When the Cluster API MachineSet has been created independently of Machine API, new Machines will not be created in Machine API to reflect the Machines owned by the Cluster API MachineSet.
+
+#### Deletion of Machines
+
+To enable the synchronisation of deletion, the sync controller will use its own Finalizer `snyc.machine.openshift.io/finalizer` on resources that have Machine API and Cluster API equivalents.
+
+This means that for a Machine, we expect the authoritative controller to perform the actual deletion work within the infrastructure provider.
+Once it has completed this effort, it will remove its own Finalizer.
+
+As soon as the synchronisation controller notices the Machine has been deleted, it should ensure that the Machine API/Cluster API equivalent (if present) is also deleted.
+It should then wait until the authoritative Machine controller removes its finalizer before removing both its own, and the Machine API/Cluster API Machine finalizer if present.
+In scenarios where both Machine API and Cluster API have been authoritative, it is expected that both Machine controllers will have added their own Finalizers, therefore we expect the synchronisation controller should be able to handle this.
+
+It is also feasible that a customer may want to remove the Machine API resources after they have migrated to Cluster API and no longer require the Machine API synchronisation.
+To allow for this, if a non-authoritative Machine API parent[^8] resource is deleted, the deletion event will not be synchronised to the Cluster API equivalent and the synchronisation controller will ensure finalizers are removed as appropriate.
```

Comment:
Could this be error prone or confusing to users? They might issue a delete request to the non-authoritative API in this case and expect the Machine to be delete instead?

An alternative we could consider is to have a cluster level or sync controller level flag to disable any conversion of sorts and only allow that whenever all resources have been converted